Extensions/Share/ShareViewController.swift

```diff
@@ -0,0 +1,312 @@
+                make.width.equalTo(14)
+            }
+        }
+
+        let gesture = UITapGestureRecognizer(target: self, action: action)
+        row.addGestureRecognizer(gesture)
+
+
+    }
+
+    fileprivate func animateToActionDoneView(withTitle title: String = "") {
+        navigationItem.leftBarButtonItem = nil
+
+        navigationController?.view.snp.updateConstraints { make in
+            make.height.equalTo(UX.viewHeightForDoneState)
+        }
+
+        actionDoneRow.label.text = title
+
+        UIView.animate(withDuration: UX.doneDialogAnimationDuration, animations: {
+            self.actionDoneRow.row.isHidden = false
+            self.stackView.arrangedSubviews.forEach { view in
+                if !self.viewsShownDuringDoneAnimation.contains(view) {
+                    view.removeFromSuperview()
+                }
+            }
+
+            self.navigationController?.view.superview?.layoutIfNeeded()
+        }, completion: { _ in
+        })
+    }
+
+    @objc func finish(afterDelay: TimeInterval = UX.durationToShowDoneDialog) {
+        delegate?.finish(afterDelay: afterDelay)
+    }
+
+    private func makeActionDoneRow(addTo parent: UIStackView) -> (row: UIStackView, label: UILabel) {
+        let stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.addBackground(color: UX.doneLabelBackgroundColor)
+        stackView.rightLeftEdges(inset: UX.rowInset)
+        parent.addArrangedSubview(stackView)
+
+        stackView.snp.makeConstraints { make in
+            make.height.equalTo(UX.pageInfoRowHeight)
+        }
+
+        let label = UILabel()
+        label.font = UX.doneLabelFont
+        label.handleLongLabels()
+
+        let checkmark = UILabel()
+        checkmark.text = "✓"
+        checkmark.font = UIFont.boldSystemFont(ofSize: 22)
+
+        [label, checkmark].forEach {
+            stackView.addArrangedSubview($0)
+            $0.textColor = .white
+        }
+
+        checkmark.snp.makeConstraints { make in
+            make.width.equalTo(20)
+        }
+
+        return (stackView, label)
+    }
+
+    private func setupNavBar() {
+        navigationController?.navigationBar.isTranslucent = false
+        navigationController?.navigationBar.setValue(true, forKey: "hidesShadow") // hide separator line
+        navigationItem.titleView = UIImageView(image: UIImage(named: "Icon-Small"))
+        navigationItem.titleView?.contentMode = .scaleAspectFit
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: Strings.SendToCancelButton, style: .plain, target: self, action: #selector(finish))
+    }
+
+    private func setupStackView() {
+        stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.spacing = 4
+        view.addSubview(stackView)
+        stackView.snp.makeConstraints { make in
+            make.edges.equalToSuperview()
+        }
+    }
+
+    override func viewDidLoad() {
```

Comment:
Reading the file top to bottom. I think it helps having the `viewDidLoad` method right after the variable declarations.